library/src/main/java/com/schibsted/spain/barista/BaristaSwipeRefreshActions.java

```diff
@@ -0,0 +1,17 @@
+package com.schibsted.spain.barista;
+
+import android.support.annotation.IdRes;
+import com.schibsted.spain.barista.custom.CustomConstraintViewAction;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.swipeDown;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayingAtLeast;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+public class BaristaSwipeRefreshActions {
+
+  public static void pullToRefresh(@IdRes int swipeRefreshId) {
+    onView(withId(swipeRefreshId)).perform(
+        new CustomConstraintViewAction(swipeDown(), isDisplayingAtLeast(80)));
```

Comment:
Another idea: If this is only supposed to work with the `SwipeRefreshLayout`, we might add a constraint for that type. We have a similar case to this in `AutocompleteViewActions`. The implementation would be something like:
```java
    @Override
    public Matcher<View> getConstraints() {
      return allOf(isDisplayingAtLeast(80), isAssignableFrom(SwipeRefreshLayout.class));
    }
```
This way Espresso will validate the view type and inform if the user is using the action with a different view by mistake.
What do you think?